server/graphql/schema/ingestion/resolvers/mutations/ingestData.ts

```diff
@@ -547,52 +547,124 @@ class IngestDataWorker extends ResolverBase {
-            Name: photogrammetry.name,
-            idVCaptureMethod: vocabulary.idVocabulary,
-            DateCaptured: H.Helpers.convertStringToDate(photogrammetry.dateCaptured) || new Date(),
-            Description: photogrammetry.description,
-            idAssetThumbnail: null,
-            idCaptureData: 0
-        });
-        if (!await captureDataDB.create()) {
-            LOG.error(`ingestData unable to create CaptureData for photogrammetry data ${JSON.stringify(photogrammetry)}`, LOG.LS.eGQL);
+        let idCaptureData: number = 0;
+        if (photogrammetry.idAsset) {
+            const asset: DBAPI.Asset | null = await DBAPI.Asset.fetch(photogrammetry.idAsset);
+            if (!asset) {
+                LOG.error(`ingestData createPhotogrammetryObjects unable to fetch asset from ${JSON.stringify(photogrammetry, H.Helpers.saferStringify)}, idAsset ${photogrammetry.idAsset}`, LOG.LS.eGQL);
+                return false;
+            }
+
+            const assetType: COMMON.eVocabularyID | undefined = await asset.assetType();
+            if (assetType === COMMON.eVocabularyID.eAssetAssetTypeCaptureDataFile ||
+                assetType === COMMON.eVocabularyID.eAssetAssetTypeCaptureDataSetPhotogrammetry) {
+                const SO: DBAPI.SystemObject | null = asset.idSystemObject ? await DBAPI.SystemObject.fetch(asset.idSystemObject) : null;
+                if (!SO) {
+                    LOG.error(`ingestData createPhotogrammetryObjects unable to fetch photogrammetry's asset's system object ${JSON.stringify(asset, H.Helpers.saferStringify)}`, LOG.LS.eGQL);
+                    return false;
+                }
+                if (SO.idCaptureData)                   // Is this a CD - Photo?
+                    idCaptureData = SO.idCaptureData;   // Yes: Use it!
+            }
+        }
+
+        let CDDB: DBAPI.CaptureData | null = idCaptureData ? await DBAPI.CaptureData.fetch(idCaptureData) : null;
+        if (CDDB) {
+            CDDB.Name = photogrammetry.name;
+            if (H.Helpers.convertStringToDate(photogrammetry.dateCaptured) instanceof Date) CDDB.DateCaptured = H.Helpers.convertStringToDate(photogrammetry.dateCaptured) as Date;
+            CDDB.Description = photogrammetry.description;
+        } else {
+            CDDB = new DBAPI.CaptureData({
+                Name: photogrammetry.name,
+                idVCaptureMethod: vocabulary.idVocabulary,
+                DateCaptured: H.Helpers.convertStringToDate(photogrammetry.dateCaptured) || new Date(),
+                Description: photogrammetry.description,
+                idAssetThumbnail: null,
+                idCaptureData: 0
+            });
+        }
+        const CDDBRes: boolean = idCaptureData ? await CDDB.update() : await CDDB.create();
+        if (!CDDBRes) {
+            LOG.error(`ingestData unable to ${idCaptureData ? 'update' : 'create'} CaptureData for photogrammetry data ${JSON.stringify(photogrammetry)}`, LOG.LS.eGQL);
```

Comment:
Use `H.Helpers.JSONStringify` in place of `JSON.stringify` in order to safely stringify bigints, Sets, and Maps.